junit5/src/main/java/cz/xtf/junit5/extensions/helpers/ResourcesPrinterHelper.java

```diff
@@ -0,0 +1,171 @@
+package cz.xtf.junit5.extensions.helpers;
+
+
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.Event;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.apps.StatefulSet;
+import io.fabric8.openshift.api.model.Build;
+import io.fabric8.openshift.api.model.BuildConfig;
+import io.fabric8.openshift.api.model.DeploymentConfig;
+import io.fabric8.openshift.api.model.ImageStream;
+import io.fabric8.openshift.api.model.Route;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class ResourcesPrinterHelper<X> implements AutoCloseable {
+	private final Path file;
+	private final Function<X, String[]> resourceToCols;
+	private final int[] maxLengths;
+	private final List<String[]> rows;
+
+	private ResourcesPrinterHelper(Path file, Function<X, String[]> resourceToCols, String... colNames) {
```

Comment:
I think the `colNames` are strictly bound to the given resource. For that reason I don't like the need to name them explicitly in the static factory methods like `forEvents`. Could the `resourceToCols` return a map instead of an array? The map would map the `colNames` to the rows which are now provided by `resourceToCols`. WDYT? This is not a strict request for change.

Reply:
Nice catch! It needs some logic (`colNames` are treated as row just like result of `resourceToCols`). But in that way (your) it will be easier to add more info.